net/net.go

```diff
@@ -16,7 +16,72 @@ limitations under the License.
 
 package net
 
-import "net"
+import (
+	"fmt"
+	"net"
+)
+
+// ParseCIDRs parses a list of cidrs and return error if any is in valid.
+// order is maintained
+func ParseCIDRs(cidrsString []string) ([]*net.IPNet, error) {
+	cidrs := make([]*net.IPNet, 0)
+	for idx, current := range cidrsString {
+		_, cidr, err := net.ParseCIDR(current)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cidr at index:%v with error:%v", idx, err)
+		}
+		cidrs = append(cidrs, cidr)
+	}
+
+	return cidrs, nil
+}
+
+// IsDualStack returns if ips are <v4>,<v6> or <v6>,<v4>
+func IsDualStackIPs(ips []net.IP) bool {
```

Comment:
Would it make sense to have this as 'IPsByFamily' and return a map of IP family to slice of IP?  That seems like a primitive we could reuse?

Reply:
Ditto for string beliw

Reply:
And for CIDR and CIDR string

Reply:
The map[<family>]<cidr> has no use in k/k code base as far as i can tell. 
`ipam controller`: assign from cidr, no dependency on family (does depend on order of cidrs) 
`route controller`: creates routes on cidrs, no dependency on family
`kubenet`: only care about cidr ranges, only cares about family when it does traffic shaping.